### PR TITLE
Enable Mypy Checking in torch/_inductor/fx_passes/split_cat.py

### PR DESCRIPTION
**Pull request by Cyril-Anto**
Fixes #105230 Enable Mypy Checking in torch/_inductor

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @Xia-Weiwen @ngimel

### CODE REVIEW

File: torch/_inductor/fx_passes/split_cat.py [hunk 1]
```diff
@@ -454,10 +454,10 @@ def get_simplified_split_ranges(
             split_ranges,
         ):  # This need not be a strict condition
             # However, we keep it now for simplicity.
-            return
+            return # type: ignore[return-value]
```
Comment (Skylion007) [inline torch/_inductor/fx_passes/split_cat.py:457]: This does seem like an actual bug in our typing logic. Should be equivalent to returning None

File: torch/_inductor/fx_passes/split_cat.py [hunk 2]
```diff
@@ -738,7 +737,7 @@ def get_simplified_split_ranges(
             split_sections, next_users, user_inputs_list
         )
         if not simplified_split_ranges or len(simplified_split_ranges) != 1:
-            return None
+            return None # type: ignore[return-value]
```
Comment (Skylion007) [inline torch/_inductor/fx_passes/split_cat.py:740]: We probably need to mark the return type as an Optional here too

**Comment by Cyril-Anto** [comment]
@Skylion007 Kindly review the changes made.

**Comment by Cyril-Anto** [comment]
@Skylion007 I've made the changes based on your suggestions, and can you please review them again?

File: torch/_inductor/fx_passes/split_cat.py [hunk 3]
```diff
@@ -454,10 +454,10 @@ def get_simplified_split_ranges(
             split_ranges,
         ):  # This need not be a strict condition
             # However, we keep it now for simplicity.
-            return
+            return None
         split_ranges = self.fill_gaps(split_ranges, 0, cumulative_sizes[-1])
         if len(split_sections) == len(split_ranges):  # Simplification not possible
-            return
+            return # type: ignore[return-value]
```
Comment (Skylion007) [inline torch/_inductor/fx_passes/split_cat.py:460]: Should be return None here too?

File: torch/_inductor/fx_passes/split_cat.py [hunk 4]
```diff
@@ -513,24 +512,24 @@ def get_transform_params(
                     ]
                     # All sections should be equal
                     if len(set(subset_split_sections)) != 1:
-                        return
+                        return # type: ignore[return-value]
```
Comment (Skylion007) [inline torch/_inductor/fx_passes/split_cat.py:515]: And here

File: torch/_inductor/fx_passes/split_cat.py [hunk 5]
```diff
@@ -733,7 +732,7 @@ def get_simplified_split_ranges(
         split_sections,
         next_users,
         user_inputs_list: List[List[Union[torch.fx.Node, Tuple[int, int]]]],
-    ) -> List[Tuple[int, int]]:
+    ) -> Optional[List[Tuple[int, int]]]:
```
Comment (Skylion007) [inline torch/_inductor/fx_passes/split_cat.py:735]: The return type here just looks entirely wrong and needs to be changed.

**Review by Skylion007 — CHANGES_REQUESTED** [review]
The return types for these functions need to be fixed.

**Comment by Cyril-Anto** [comment]
@Skylion007 I have made the changes you suggested, Can you please do review it once?

**Pull request closed without merging**